### PR TITLE
doc work

### PR DESCRIPTION
**Pull request by thekemkid**
Adds an example, as well as basic readme content

### CODE REVIEW

**Comment by mcollina** [comment]
Would you mind splitting the server and the anger stuff in the example?

**Comment by thekemkid** [comment]
I kept them in the same place so you could see the setting of the payload meta id that corresponds to a request. I'll split if you want though, but I thought it makes it easier for new users to see that in the same file.

Splitting _could_ just be requiring buildServer from ./test/server.js, as that is where I took it from.

**Comment by mcollina** [comment]
The whole point of having them split is to emulate a real world scenario. When you use this tool, you don't run the server in the same process as the benchmarker: we do that in tests, because that's convenient, and we don't care about real numbers. 

And don't load it up from test, it needs to be separate and self sustaining. Examples should be something that can be run by copy-pasting them (spare the `require('..')` change).

**Comment by thekemkid** [comment]
Okay, that makes perfect sense. updating.

**Comment by mcollina** [comment]
Can you do one examples folder, and call them "benchmark" and "server"?

**Comment by mcollina** [comment]
And maybe add a little readme in there as well?

File: README.md [hunk 1]
```diff
@@ -1,4 +1,83 @@
...
+* `error`: emitted on a error. The callback function will receive the `error` as the first parameter.
+* `connect`: emitted when all clients are connected to the server.
+* `subscribe`: emitted when all clients are subscribed to the server.
+* `trigger`: emitted when a sender is sending a 'request' message. The callback function will receive the requests `uid` as the first parameter.
+* `publish-events-recieved`: emitted when all connections receive the message 'response' sent from the server, in response to some triggered 'request'. The callback should receive the `uid` for the event that triggered it.
+* `end`: emmited when testing finishes, with the `result` passed as the first parameter to the callback.
+
+## Sample
+
+Check out the examples folder for a simple sample.
```
Comment (mcollina) [inline README.md]: You should link the folder

File: examples/readme.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Anger Samples
+
+Checkout [benchmark.js](./benchmark.js) to see an example of this in use with an [example nes server](./server.js).
+
+Note how the server is setup to handle response messages by assigning responses the id in the request.
+
+The benchmark.js file is a simple benchmark against an example pub/sub server.
```
Comment (mcollina) [inline examples/readme.md]: This is missing how to start the two things.

Comment (thekemkid) [inline examples/readme.md]: fixed

File: examples/server.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const hapi = require('hapi')
```
Comment (mcollina) [inline examples/server.js]: use strict missing.

**Comment by mcollina** [comment]
Thanks, great job!

**Pull request merged**